state/remote/artifactory_test.go

```diff
@@ -53,3 +56,80 @@ func TestArtifactoryFactory(t *testing.T) {
 		t.Fatalf("Incorrect subpath was populated")
 	}
 }
+
+func TestArtifactoryFactoryEnvironmentConfig(t *testing.T) {
+	// This test just instantiates the client. Shouldn't make any actual
+	// requests nor incur any costs.
+
+	config := make(map[string]string)
+
+	testArtifactoryVar(t, "username")
+	testArtifactoryVar(t, "password")
+	testArtifactoryVar(t, "url")
+	testArtifactoryVar(t, "repo")
+	testArtifactoryVar(t, "subpath")
+
+	os.Setenv("ARTIFACTORY_USERNAME", "test")
```

Comment:
Hi @juicemia,

I just read your great [blog post](https://juicemia.com/post/contributing-oss/)!

I really like when you document the refactoring, and I think that you could go even further here, using a simple helper function:

```go
// tempEnv set the environment variable
// and returns a function to reverse this change (to be used with defer)
func tempEnv(key, value string) func() {
	previous, ok := os.LookupEnv(key)
	os.Setenv(key, value)
	return func() {
		if ok {
			os.Setenv(key, previous)
		} else {
			os.Unsetenv(key)
		}
	}
}
```

Usage, instead of `os.Setenv` + `os.Unsetenv`:
```go
defer tempEnv("ARTIFACTORY_USERNAME", "test")()
```

What do you think?

Reply:
I like this better because it really sets the state of the environment to what it was before, instead of just assuming that there was nothing there. I'll update the PR as soon as I get a chance.